recipes/flatcc/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+class FlatccConan(ConanFile):
+    name = "flatcc"
+    description = "C language binding for Flatbuffers, an efficient cross platform serialization library"
+    topics = "conan", "flatbuffers", "serialization"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/dvidelabs/flatcc"
+    license = "Apache-2.0"
+    options = { "shared": [True, False],
+                "fPIC": [True, False],
+                "portable": [True, False],
+                "gnu_posix_memalign": [True, False],
+                "runtime_lib_only": [True, False],
+                "verify_assert": [True, False],
+                "verify_trace": [True, False],
+                "reflection": [True, False],
+                "native_optim": [True, False],
+                "fast_double": [True, False],
+                "ignore_const_condition": [True, False],
+    }
+    default_options = { "shared": True,
+                        "fPIC": True,
+                        "portable": False,
+                        "gnu_posix_memalign": True,
+                        "runtime_lib_only": False,
+                        "verify_assert": False,
+                        "verify_trace": False,
+                        "reflection": True,
+                        "native_optim": False,
+                        "fast_double": False,
+                        "ignore_const_condition": False
+    }
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+
+    _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```

Comment:
```suggestion
    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.settings.os == "Windows":
```

Keep it separated

Reply:
Correct.
But 
```
if self.options.shared:
            del self.options.fPIC
```
should stay in configure? For consistency I would assume I have to move those lines also config_options.

Reply:
No, this has to stay in `configure` because it depends on a option.

conan evaluates the options as follows:

1. run `config_options`
2. parse the options from the arguments
3. run `configure`

Since `options.shared` is only filled in step 2, this has to stay in `configure.

Reply:
@madebr Thanks for the explanation.